addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -2197,13 +2197,13 @@ export class Wysiwyg extends Component {
                 const bannerElement = parseHTML(`
                     <div class="o_editor_banner o_not_editable lh-1 d-flex align-items-center alert alert-${alertClass} pb-0 pt-3" role="status" data-oe-protected="true">
                         <i class="fs-4 fa ${iconClass} mb-3" aria-label="${_t(title)}"></i>
-                        <div class="w-100 ms-3" data-oe-protected="false">
+                        <div class="w-100 ms-3" data-oe-protected="false" contenteditable="true">
```

Comment:
Why is this needed? I don't see anything making the banner not editable in the first place, and we sure don't want this to remain editable when the editor is in readonly mode.
If it's just to fix the Powerbox hint, it's the wrong way to fix it. Is it possible to edit the paragraph before your fix?

Reply:
@Zynton I have implemented changes to ensure that when a banner is created, the cursor will now appear inside the banner. Previously, before the merge of this [PR](https://github.com/odoo/odoo/pull/143362/files) the cursor automatically appeared inside the banner upon its creation. However, this functionality was affected after the removal of 'o_editable.'

Regarding the issue related to the powerbox hint, the problem arose because the selection was set on the 'div' element instead of the 'p' tag.

If we dont want the selection inside the banner when it is created, then their is no need for this changes. Additionally, the line of code 'setSelection(bannerElement.querySelector('.o_editor_banner > div'), 0);' is no longer necessary, as it was responsible for setting the selection inside the banner. Therefore, this line can be removed

Reply:
Would it work if you kept your `setSelection` change and removed the `contenteditable="true"` ? I would expect it to if the banner is in the root editable. If not, how come?

Reply:
`<div class="o_editor_banner o_not_editable lh-1 d-flex align-items-center alert alert-${alertClass} pb-0 pt-3" role="status" data-oe-protected="true">` is `contentEditable=False` and because of that `setSelection` will not set selection within the `p` tag.
It seems that `o_not_editable` class is responsible for setting `contentEditable` to false.

Reply:
Ah thank you I hadn't noticed the `.o_not_editable` on the banner!
The [PR you linked to](https://github.com/odoo/odoo/pull/143362) explicitly adds the `div` to which you applied `contenteditable="true"` to the array returned by `getContentEditableAreas`. Then, `_activateContenteditable` is supposed to apply `contenteditable="true"` to it automatically (and `_stopContenteditable` is supposed to remove it so you wouldn't still have it in readonly). So your code should not be needed. Instead, we need to figure out why `_activateContenteditable` doesn't apply `contenteditable="true"`. When it calls `getContentEditableAreas`, does it indeed receive the banner's div as part of the response? I see the PR has a big comment above `getContentEditableAreas`:
```js
// TODO other places in this file call this.options.getContentEditableAreas
// without the extension here. It does not seem to be a problem (it
// was like that before o_editor_banner elements were considered
// here), but we might want to review that.
```
It seems that this would be the most obvious cause. If `_activateContenteditable` doesn't receive the banner's div as part of the response from `getContentEditableAreas`, then `getContentEditableAreas` is called without the needed extension, and that's what needs fixing.

Reply:
In a typical scenario, `_activateContenteditable` is called on `_onMouseDown` or `historyRevertCurrentStep`. However, when creating a new banner, neither of these methods is invoked. Consequently, `_activateContenteditable` is not called, and the banner remains non-editable upon creation. To address this, I have call `_activateContenteditable` after the banner is inserted to ensure it make the banner editable.